sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -54,8 +55,11 @@ private[spark] object HiveUtils extends Logging {
     sc
   }
 
+  private val supportedHive2ShortVersions = Set("2.3.0")
+  val isSupportedHive2 = supportedHive2ShortVersions.contains(HiveVersionInfo.getShortVersion)
+
   /** The version of hive used internally by Spark SQL. */
-  val builtinHiveVersion: String = "1.2.1"
+  val builtinHiveVersion: String = if (isSupportedHive2) HiveVersionInfo.getVersion else "1.2.1"
```

Comment:
If we've removed the dependency on our fork of 1.2.1, when would we ever not have Hive 2.x built in?

Reply:
@srowen this change doesn't remove the dependency on our fork of 1.2.1 -- see

https://github.com/apache/spark/blob/eaf008ad0e6e246665127c283b139a16424f3139/pom.xml#L127-L130

https://github.com/apache/spark/blob/eaf008ad0e6e246665127c283b139a16424f3139/pom.xml#L1384-L1386

Or are you proposing including a bigger change here to do that as well?

Reply:
Yeah, I guess don't do it here, but I feel like it's inevitable we have to remove this fork. It doesn't mean it can't interoperate with Hive metastores in older versions, right?